packages/react-native-codegen/src/parsers/parsers-commons.js

```diff
@@ -898,6 +903,79 @@ function getEventArgument(
   };
 }
 
+function getTypeResolutionStatus(
+  type: 'alias' | 'enum',
+  typeAnnotation: $FlowFixMe,
+  parser: Parser,
+): TypeResolutionStatus {
+  return {
+    successful: true,
+    type,
+    name: parser.nameForGenericTypeAnnotation(typeAnnotation),
+  };
+}
+
+function handleGenericTypeAnnotation(
+  typeAnnotation: $FlowFixMe,
+  resolvedTypeAnnotation: TypeDeclarationMap,
+  parser: Parser,
+): {
+  typeAnnotation: $FlowFixMe,
+  typeResolutionStatus: TypeResolutionStatus,
+} {
+  const parserType = parser.language();
+  let typeResolutionStatus;
+  let node;
+
+  switch (resolvedTypeAnnotation.type) {
+    case parser.typeAlias: {
+      typeResolutionStatus = getTypeResolutionStatus(
+        'alias',
+        typeAnnotation,
+        parser,
+      );
+      node =
+        parserType === 'Flow'
+          ? resolvedTypeAnnotation.right
+          : resolvedTypeAnnotation.typeAnnotation;
+      break;
+    }
+    case parser.enumDeclaration: {
+      typeResolutionStatus = getTypeResolutionStatus(
+        'enum',
+        typeAnnotation,
+        parser,
+      );
+      node =
+        parserType === 'Flow'
+          ? resolvedTypeAnnotation.body
+          : resolvedTypeAnnotation;
+      break;
+    }
+    case 'TSInterfaceDeclaration': {
```

Comment:
I didn't use parser.interfaceDeclaration here because in Flow, it should fall through to the default switch case.

Reply:
Can you maybe add a comment explaining that? Otherwise, in a few weeks we would think about why we left `TSInterfaceDeclaration` here